modules/core/src/main/java/org/apache/ignite/internal/util/StripedExecutor.java

```diff
@@ -139,20 +143,28 @@ public StripedExecutor(
     }
 
     /**
-     * Checks starvation in striped pool. Maybe too verbose
-     * but this is needed to faster debug possible issues.
+     * Checks starvation in striped pool.
      */
     public void checkStarvation() {
-        for (int i = 0; i < stripes.length; i++) {
-            Stripe stripe = stripes[i];
+        isStarvationDetected();
+    }
 
-            long completedCnt = stripe.completedCnt;
+    /**
+     * Check starvation in striped pool. Maybe too verbose
+     * but this is needed to faster debug possible issues.
+     *
+     * @return Flag representing presence of possible starvation in striped pool.
+     */
+    public boolean isStarvationDetected() {
+        boolean retVal = false;
 
+        for (Stripe stripe : stripes) {
             boolean active = stripe.active;
 
-            if (completedCntrs[i] != -1 &&
-                completedCntrs[i] == completedCnt &&
-                active) {
+            long lastStartedTs = stripe.lastStartedTs;
+
+            if (lastStartedTs + threshold < U.currentTimeMillis() && active) {
+                retVal = true;
```

Comment:
can you rename it to something descriptive, such as 'starvation'?

Reply:
can you also put `active` boolean check first? I.e. if (active && ...

Reply:
ok